### DISCUSSION TITLE
安装插件或带有插件触发器的插件失效的解决方案[已解决]

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2342] [重要通知]
>  ### 简悦 2.3.0 版彻底解决了插件各种问题（包括此问题）

# 🎉 简悦已更新到 2.3.0 版，适配 Manifest V3 方案，已上架 Chrome 应用商店

# 📥 Chrome 应用商店地址

https://chromewebstore.google.com/detail/simpread-reader-view/ijllcpnolfcooahcekpamkbidhejabll

# 🔗 讨论地址

https://github.com/Kenshin/simpread/issues/7326

# 🔌 插件系统 2.0

插件系统是这两个版本变化最大的地方，因为 Manifest V3 完全禁止了远超加载代码，导致插件系统改为「内置」方案。

2.2.0 → 在插件中心查看/安装插件
2.3.0 → 在简悦内启用/禁用插件（没有安装/卸载的功能）

关于插件系统 2.0 内置插件的更多介绍 https://github.com/Kenshin/simpread/issues/7329

# 🟩 带来的好处

- 当你安装插件，进入阅读模式后没有在「插件触发器」找到对应插件的问题。
- 更换为新设备时，插件配置无故「消失」的问题。
- 插件保存后但并没有生效的问题。
- 无法安装插件的问题。
- 导入配置文件后，但没有看到插件的问题。

***
***
***
***

# 问题现象

安装 [阅读模式浮动面板定制器](https://simpread.ksria.cn/plugins/details/7gcXYY4HAc) 如果没有出现 **插件触发器** 或 **插件配置保存后不生效**。

**没有插件触发器**

![QQ截图20210606182021](https://user-images.githubusercontent.com/38481514/120921023-54aee500-c6f4-11eb-8480-2c60894cf05d.png)

**已经安装了插件**

![image](https://user-images.githubusercontent.com/38481514/120921061-78722b00-c6f4-11eb-986f-a0422a190b1e.png)

或插件配置保存后不生效。

# 问题原因

使用 **同步助手的自动同步功能** 造成的，可能在安装后没有保存到同步助手对应的本地配置文件。

# 解决方案

简悦近期增加了 [插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 可以通过此插件彻底解决此问题，可直接看此[一站式教程](https://www.yuque.com/kenshin/simpread/fcw60u)。

# 引申

安装站点集市的站点失效解决方案 https://github.com/Kenshin/simpread/discussions/2342

**Comment by blankwz** [comment]
我的不生效。

**Reply by Kenshin** [reply]
不会没用的，你也可以试试下面的步骤：

1. 导出配置文件
2. 重新安装简悦
3. 重启浏览器
4. 安装包含插件触发器的插件
5. 随便进入个页面 https://sspai.com/post/67904

看看插件触发器是否出现？

**Reply by blankwz** [reply]
谢谢！出现了！

**Reply by Kenshin** [reply]
搞定就好。

**Comment by 1wingedangel** [comment]
我这边解决方法1和解决方法2都不生效，覆盖配置之后就变回了原来的插件不生效的情况。

第三种解决方法看似可以，但是我一旦导入自己备份好了的配置文件就又变回去了……

如果要启用插件就必须得Full Wipe，也就意味着稍后读的数据全部都要丢弃。非常纠结。

**Reply by Kenshin** [reply]
嗨，试试这样

![image](https://user-images.githubusercontent.com/81074/132982960-b3dadbcd-42a7-406a-a3dd-c2070d1efe34.png)

注意：此方式不会删除插件中的配置信息。

***

> 稍后读的数据全部都要丢弃

只要有 `simpread_config.json` （人为备份或者通过网盘查看历史记录）数据就不会丢。

**Reply by 1wingedangel** [reply]
我再详细解释一下问题：

解决方法1：

- 先关闭同步助手 //正常
- 重新安装 此插件 //正常
- 确保可以看到 插件触发器 //正常
- 把配置文件导出到本地 → 改名为 simpread_config.json 再覆盖到同步助手对应的同名文件。 //正常
- 开启同步助手（不选覆盖）//如果不覆盖配置文件的话刷新页面，同步助手始终处于关闭状态。覆盖配置文件之后同步助手可以开启，但是插件触发器消失了。

解决方法2：

- 先关闭同步助手 //正常
- 重新安装 此插件 //正常
- 确保可以看到 插件触发器 //正常
- 开启同步助手，并覆盖当前配置到本地 //覆盖配置文件之后插件触发器消失

解决方法3：

- 导出配置文件 //正常
- 重新安装简悦 //正常
- 重启浏览器 //正常
- 安装包含插件触发器的插件 //正常
- 随便进入个页面 https://sspai.com/post/67074 //正常
- 此时导入自己的配置文件，插件触发器消失；不导入自己的配置文件的话稍后读数据就导入不进去；导入助手的数据到本地也会导致插件触发器消失。

总结下来就是：现在稍后读数据和插件是二选一的状态，要插件就没有稍后读数据，要导入稍后读数据就没有插件。以后建议是不是可以把稍后读的配置文件和简悦的配置文件给分开来，这样配置出现了兼容性问题可以保证不丢失稍后读的数据。

**Reply by Kenshin** [reply]
嗨，谢谢这么精细的测试。🙏

先回复下

> 以后建议是不是可以把稍后读的配置文件和简悦的配置文件给分开来，这样配置出现了兼容性问题可以保证不丢失稍后读的数据。

确实如此，至少会区分配置数据与稍后读数据的。

***

你的测试结果有点奇怪，你可以这么做

> 方案一

改名 `simpread_config.json` 后，可以有任意文本编辑打开它，然后检索 `remote` 字段（第一个检索结果）后，将 `"sync":false` → `"sync":true` 如下图：（下图为导出后格式化的效果

![image](https://user-images.githubusercontent.com/81074/133015369-48211598-e1bc-414f-9afd-bb89944589ac.png)

操作完，在确保选项页下图为开启状态

![image](https://user-images.githubusercontent.com/81074/133015599-d2434287-a370-423c-be5c-7815c6d8fa16.png)

最后覆盖之前的 `simpread_config.json` 刷新选项页，再看看。

***

PS：这个情况属于极少数发生的问题，而你的情况则是极少数中的少数。

**Reply by 1wingedangel** [reply]
> 改名 simpread_config.json 后，可以有任意文本编辑打开它，然后检索 remote 字段（第一个检索结果）后，将 "sync":false → "sync":true 如下图：（下图为导出后格式化的效果

好的，按照这个方法，同步确实开启了，浮动定制插件能够工作了。

但是我再加几个插件（比如 iOS Assist Touch插件），刷新页面之后，在插件栏会显示启用这个插件，但是动作栏里还是没有此插件的动作。感觉我需要再重复解决方案一才能添加其他插件了。

我感觉或许有可能是旧版本的简悦配置文件和新版本的发生了兼容性问题。如果稍后读数据能够分离出来的话我倒是想重置一下配置再重新折腾一下的……

**Reply by Kenshin** [reply]
如果愿意折腾的话，可以重置数据，留意下面截图的内容

![image](https://user-images.githubusercontent.com/81074/133037978-19ca46a9-6364-42c8-b6eb-b4529f976073.png)

尤其是 `unrdist` `collections` 都是很重要的稍后读和合辑数据。

**Reply by 1wingedangel** [reply]
问题已初步解决。经过一些测试，是由于简约同步助手（环境：Windows+Edge）在自动同步的过程中，不会把浏览器配置中的 `option > plugin` 相关的数据，和 `plugstorge` 的相关数据同步到本地，然后本地配置会在刷新配置页面后反过来覆盖浏览器配置导致浏览器配置中的相关数据也会丢失。

另，在我的环境下好像站点集市的数据也是同样的道理，安装后页面自动刷新>然后已安装的站点规则就直接消失了。

这里贴出自己折腾出来的解决方案（可以算是4了）：

* 关闭同步助手。
* 刷新选项页面。
* 安装所需插件，站点集市里安装站点规则。
* 随便打开一个测试页，进入阅读模式。对插件配置进行修改。
* 刷新测试页，进入阅读模式。检查插件配置改动是否已生效。
* 回到选项页面，刷新一次页面。
* 打开同步助手，**千万不要刷新页面**，直接点击 `通过简悦同步助手覆盖本地配置文件` 。出现确认框后再确认。
* 刷新页面，检查同步助手是不是处于开启状态。
* 随便打开测试页进入阅读模式，检查插件是否生效。

需要注意的是，如果你要对插件和站点规则进行任何改动，**每次都需要运行这个解决方案** 。希望能够作为一种参考。

**Reply by Kenshin** [reply]
备注：

此方式以及问题太过特殊，暂时并不具有共性。

如果看到这里的用户，请已官方方案为准。

**Comment by Lynyh22** [comment]
四种方案都不生效，同步一开就失效。

**Reply by Kenshin** [reply]
可以的，否则我也不能发上来。 😂

**Reply by Lynyh22** [reply]
不行啊，我再一次按照方案一严格按照每一步走，无论安装了多少个插件，一点从配置文件导入插件，最终只会留下浮动面板定制器，其他全没了......

**Reply by Lynyh22** [reply]
好像情况就跟上面那位说的一样

**Reply by 1wingedangel** [reply]
试试这个？ https://github.com/Kenshin/simpread/discussions/2342#discussioncomment-1353539

**Reply by Lynyh22** [reply]
添加了一个插件，可行。不知道一次性添加多个插件怎么样。

**Reply by Lynyh22** [reply]
> 试试这个？ [#2342 (reply in thread)](https://github.com/Kenshin/simpread/discussions/2342#discussioncomment-1353539)

试了一下，确实对插件进行变动不重新运行一遍是无法生效的......

**Reply by Kenshin** [reply]
@Lynyh22 

恭喜你遇到了最隐藏的 Bug （是不是可以叫彩蛋了... 
99% 的用户都没问题，1% 的用户通过方案一就可以，这其中的 1% 通过方案二三四就可以了，目前只遇到你们两个有此问题。

PS：下个版本会重构插件的保存机制。

**Reply by Lynyh22** [reply]
😂啊这…没办法只能暂时不用插件了，而且刚刚看了一下，导入Obsidian的功能好像也有问题

**Reply by 1wingedangel** [reply]
就是100%的Windows用户没错了(•‿•)

**Reply by Kenshin** [reply]
> 😂啊这…没办法只能暂时不用插件了，而且刚刚看了一下，导入Obsidian的功能好像也有问题

其实，插件并不会太频繁更新，所以你可以使用之前的办法，而且并不是 100% Windows 用户都有此问题。 😂  

PS：彩蛋无疑了~

**Comment by NewEarthTask** [comment]
1. 插件没反应，第一步必须关闭同步助手，基础设置中也关掉，装的同步助手也停止同步。然后打开插件中心，清空本地插件，找一个以前没用过的比如分栏插件，下载，刷新，打开一个新网页，阅读模式，会弹出来插件框和触发器。然后你再下载你以前用的插件，再打开一个新网页，阅读模式，这时候虽然不弹窗，但是都会有触发器，然后你开就能正常用了。

2. 确实以前下载过的插件，除了网页风格什么的，比如悬浮面板，liveediter，再重新装的时候，不弹窗不显示触发器。我也出现了这个问题好几次，因为我 把我的简悦搞乱了，卸载安装设置之类好多次，还买了三四个简悦账号，终于按照教程把全平台联动Notion、flome、坚果云等等全打通了。出现的错都自己捣鼓出来了。嗯嗯建议问主推广简悦，我在少数派评价里面，推荐软件全部选择的其他，然后写简悦、flomo、Notion好评不解释。简悦好！！！

3. 简悦真好，因为是一个人开发的，大佬挺辛苦不容易，以后肯定搞得更方便舒服了，加油加油，再说一次，简悦真棒我真喜欢。哈哈哈哈哈哈哈

**Reply by Kenshin** [reply]
哈哈，谢谢对简悦的喜爱。

你的解决方案我已经放到了官方提示方案里面。🙏   👍 

> 我在少数派评价里面，推荐软件全部选择的其他，然后写简悦、flomo、Notion好评不解释。简悦好！！！

🎉

**Comment by NewEarthTask** [comment]
哈哈哈哈哈哈，喜爱加钦佩，用着开心👏👏👍👍

**Comment by Kenshin** [comment]
嗨，[插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 更新到 2.0.0 

## 彻底解决

1️⃣ 无法安装插件

2️⃣ 安装插件后插件的配置保存不生效

的问题。

## 一站式教程（含下载地址）

https://www.yuque.com/kenshin/simpread/fcw60u

## 注意

并不是【用同步助手的自动同步功能】后 100% 无法安装插件，如果你遇到了此问题，可以使用此插件解决。

## 视频操作

https://user-images.githubusercontent.com/81074/180121389-550ccfbc-52b3-4d17-996c-125071746a26.mp4
